docs/get-details/dapps/smart-contracts/apps/index.md

```diff
@@ -452,7 +452,73 @@ Additionally, when validating transactions, using relative position of transacti
+* Deletion: 
+    * Deleting an app does not affect its local storage. 
+    * _Clear state_. Every Smart Contract on Algorand has two programs: the _approval_ and the _clear state_ program. An account holder can clear their local state for an app at any time (deleting their data and freeing up their locked minimum balance). The purpose of the clear state program is to allow the app to handle the clearing of that local state gracefully. 
+    * Account x can request to clear its local state using a [close out transaction](https://developer.algorand.org/docs/get-details/transactions/#application-close-out-transaction). 
+    * Account x can clear its local state for app a using a [clear state transaction](https://developer.algorand.org/docs/get-details/transactions/#application-clear-state-transaction), which will always succeed, even after app a is deleted. 
+
+
+**Boxes for app a:** 
+
+* Allocation: 
+    * App a can allocate as many boxes as it needs, when it needs them.
+    * App a allocates a box using the box_create opcode in its TEAL program, specifying the name and the size of the box being allocated. 
+        * Boxes can be any size from 0 to 32K bytes. 
+        * Box names must be at least 1 byte, at most 64 bytes, and must be unique within app a. 
+    * The app account(the smart contract) is responsible for funding the box storage (with an increase to its minimum balance requirement, see below for details). 
+    * A box name must be referenced in the boxes array of the app call to be allocated. 
+* Reading: 
+    * App a is the only app that can read the contents of its boxes on-chain. This on-chain privacy is unique to box storage. Recall that everything can be read by anybody from off-chain using the algod or indexer APIs. 
+    * To read box b from app a, the app call must include b in its boxes array. 
+    * Read budget: Each box reference in the boxes array allows an app call to access 1K bytes of box state - 1K of “box read budget”. To read a box larger than 1K, multiple box references must be put in the boxes arrays. 
+        * The box read budget is shared across the transaction group. 
+        * The total box read budget must be larger than the sum of the sizes of all the individual boxes referenced (it is not possible to use this read budget for a part of a box - the whole box is read in).
+    * Box data is unstructured. This is unique to box storage. 
+    * A box is referenced by including its app ID and box name. 
+* Writing: 
+    * App a is the only app that can write the contents of its boxes.
+    * Exactly analogous to reading, each box ref in the boxes array allows an app call to write 1K bytes of box state - 1K of “box write budget”. 
+* Deletion: 
+    * App a is the only app that can delete its boxes. 
+    * If an app is deleted, its boxes are not deleted. (the correct cleanup design is to look up the boxes from off-chain and call the app to delete all its boxes before deleting the app itself). 
```

Comment:
aren't they deleted? the min balance would be locked if there are boxes outstanding but i thought the boxes themselves would also be gone

Reply:
more a clarification, are the boxes indeed deleted? or do they remain possible to query from the algod api?